plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java

```diff
@@ -229,6 +235,14 @@ public PinotConfig setCountDistinctPushdownEnabled(boolean countDistinctPushdown
         return this;
     }
 
+    @Config("pinot.json-predicate-pushdown.enabled")
```

Comment:
Why do we need this toggle?

Reply:
> The new session property (pinot.json_predicate_pushdown_enabled) defaults to false because JSON_MATCH can fail when a JSON index hasn't been configured in Pinot.

Is it possible to check the index config during analyze?

Reply:
There isn't a way to determine if the JSON index has been completely applied. Retrieving the table config from the Pinot Controller API (`/tables/{tableName}`) and inspecting `tableIndexConfig.jsonIndexColumns` would only confirm that it has been configured and the query could still fail if the JSON index was recently added and the table's segments haven't been reloaded.

Reply:
@ebyhr Assuming a missing JSON index rarely happens (or only during development), I propose we change the default to true. Do you agree?